src/Controls/src/Core/NavigationPage/NavigationPage.cs

```diff
@@ -575,14 +575,31 @@ void OnWindowChanged(object sender, EventArgs e)
 				else
 				{
 					// Is the root the window or is this part of a modal stack
-					var rootPage = this.FindParentWith(x => (x is IWindow te || Window.Navigation.ModalStack.Contains(x)), true);
+					Element toolbarRoot;
 
-					if (rootPage is Window w)
+					var parentPages = this.GetParentPages();
+					parentPages.Insert(0, this);
```

Comment:
Would it be possible to hit an NRE here if there are no parent pages? Maybe an NRE could happen if you're not using shell and have a NavigationPage as the main page ...right?

I lack enough knowledge on this topic, but I'm curious.

Reply:
From what I can tell `GetParentPages` will always return something

https://github.com/dotnet/maui/blob/095a94ea532e358c275743cff071ea779f1c1221/src/Controls/src/Core/ViewExtensions.cs#L363-L374